directiveAngular/src/app/profilemaking/profilemaking.component.html

```diff
@@ -0,0 +1,65 @@
+<form [formGroup]="ProfileForm" (ngSubmit)="onSubmit()">
+    <label for="fname">First Name: </label>
+    <input id="fname" type="text" formControlName="fname"><br>
+    
+    <div *ngIf="ProfileForm.controls.fname.errors?.['required'] && (ProfileForm.controls.fname.touched||ProfileForm.controls.fname.dirty)">
+        First name is required.
+    </div>
+    <br> <label for="lname">Last Name: </label>
+    <input id="lname" type="text" formControlName="lname"><br>
+    <label for="email">Email: </label>
+    <input id="email" type="text" formControlName="email" #emaill><br>
+   
+    <div *ngIf="ProfileForm.controls.email.errors?.['required'] && (ProfileForm.controls.email.touched||ProfileForm.controls.email.dirty)">
+        Email is required.
+    </div>
+    <div
+        *ngIf="ProfileForm.controls.email.invalid && (ProfileForm.controls.email.touched||ProfileForm.controls.email.dirty) && !ProfileForm.controls.email.errors?.['required']">Email is invalid</div>
+    <label for="phoneNumber">phoneNumber: </label>
+    <input id="phoneNumber" type="text" formControlName="phoneNumber"><br>
+    <div *ngIf="ProfileForm.controls.phoneNumber.errors?.['required'] && (ProfileForm.controls.phoneNumber.touched||ProfileForm.controls.phoneNumber.dirty)">
+        phoneNumber is required.
+    </div>
+    <div *ngIf="ProfileForm.controls.phoneNumber.errors?.['maxlength']">
+        phoneNumber maxlength is 10

```

Comment:
this is text right? so it should be like phone number must be less than 10 digits